src/features/user/Widget/EmbedModal.tsx

```diff
@@ -62,32 +64,21 @@ export default function EmbedModal({
     };
     if (contextLoaded && token) {
       try {
-        putAuthenticatedRequest(
+        const res = await putAuthenticatedRequest(
           `/app/profile`,
           bodyToSend,
           token,
-          impersonatedEmail,
-          handleError
-        )
-          .then((res) => {
-            setSeverity('success');
-            setSnackbarMessage(ready ? t('editProfile:profileSaved') : '');
-            setEmbedModalOpen(false);
-            setIsUploadingData(false);
-            setUser(res);
-          })
-          .catch((error) => {
-            setSeverity('error');
-            setSnackbarMessage(ready ? t('editProfile:profileSaveFailed') : '');
-            handleSnackbarOpen();
-            setIsUploadingData(false);
-            console.log(error);
-          });
-      } catch (e) {
-        setSeverity('error');
-        setSnackbarMessage(ready ? t('editProfile:profileSaveFailed') : '');
+          impersonatedEmail
+        );
+        setSeverity('success');
+        setSnackbarMessage(ready ? t('editProfile:profileSaved') : '');
         handleSnackbarOpen();
+        setEmbedModalOpen(false);
+        setIsUploadingData(false);
+        setUser(res);
+      } catch (err) {
         setIsUploadingData(false);
+        setErrors(handleError(err as APIError));
```

Comment:
We need a single popup that handles everything(error, warning, info, success).
This can be taken care in other PR.
Opened an Issue here https://github.com/Plant-for-the-Planet-org/planet-webapp/issues/1740